virtwho/register.py

```diff
@@ -1,5 +1,487 @@
+            if re.match(r"^Product ID:", line):
+                product_id = line.split(':')[1].strip()
+                install_attr['product_id'] = product_id
+            if re.match(r"^Version:", line):
+                version = line.split(':')[1].strip()
+                install_attr['version'] = version
+            if re.match(r"^Arch:", line):
+                arch = line.split(':')[1].strip()
+                install_attr['arch'] = arch
+            if re.match(r"^Status:", line):
+                status = line.split(':')[1].strip()
+                install_attr['status'] = status
+            if re.match(r"^Status Details:", line):
+                status_details = line.split(':')[1].strip()
+                install_attr['status_details'] = status_details
+            if re.match(r"^Starts:", line):
+                starts = line.split(':')[1].strip()
+                install_attr['starts'] = starts
+            if re.match(r"^Ends:", line):
+                ends = line.split(':')[1].strip()
+                install_attr['ends'] = ends
+        return install_attr
+
+    def sku_analyzer(self, attr):
+        """
+        Analyze the sku attributes to a dict.
+        :param attr: output get from --av or --co
+        :return: a dict including all sku attributes.
+        """
+        sku_attr = dict()
+        attr = attr.strip().split("\n")
+        for line in attr:
+            if re.match(r"^Subscription Name:", line):
+                sku_attr['sku_name'] = line.split(':')[1].strip()
+            if re.match(r"^SKU:", line):
+                sku_attr['sku_id'] = line.split(':')[1].strip()
+            if re.match(r"^Contract:", line):
+                sku_attr['contract_id'] = line.split(':')[1].strip()
+            if re.match(r"^Pool ID:", line):
+                sku_attr['pool_id'] = line.split(':')[1].strip()
+            if re.match(r"^Available:", line):
+                sku_attr['available'] = line.split(':')[1].strip()
+            if re.match(r"^Suggested:", line):
+                sku_attr['suggested'] = line.split(':')[1].strip()
+            if re.match(r"^Service Level:", line):
+                sku_attr['service_level'] = line.split(':')[1].strip()
+            if re.match(r"^Service Type:", line):
+                sku_attr['service_type'] = line.split(':')[1].strip()
+            if re.match(r"^Subscription Type:", line):
+                sku_attr['sub_type'] = line.split(':')[1].strip()
+                if 'Temporary' in sku_attr['sub_type']:
+                    sku_attr['temporary'] = True
+                else:
+                    sku_attr['temporary'] = False
+            if re.match(r"^Ends:", line):
+                sku_attr['ends'] = line.split(':')[1].strip()
+            if (re.match(r"^System Type:", line)
+                    or re.match(r"^Entitlement Type:", line)):
+                sku_attr['sku_type'] = line.split(':')[1].strip()
+        return sku_attr
```

Comment:
any common ways to transfer  the output to a dict or json?

Reply:
Done